src/libraries/Microsoft.PowerFx.Interpreter/ParsedExpression.cs

```diff
@@ -111,27 +114,29 @@ internal ParsedExpression(IntermediateNode irnode, ScopeSymbol topScope, StackDe
             // $$$ can't use current culture
             _cultureInfo = cultureInfo ?? CultureInfo.CurrentCulture;
         }
-        
+
         public async Task<FormulaValue> EvalAsync(CancellationToken cancellationToken, IRuntimeConfig runtimeConfig = null)
         {
-            ReadOnlySymbolValues symbolValues = ComposedReadOnlySymbolValues.New(
-                false,
-                _allSymbols,
-                runtimeConfig?.Values,
-                _globals);
+            ReadOnlySymbolValues symbolValues = ComposedReadOnlySymbolValues.New(false, _allSymbols, runtimeConfig?.Values, _globals);
+            BasicServiceProvider innerServices = new BasicServiceProvider();
+            bool hasInnerServices = false;
 
-            IServiceProvider innerServices = null;
             if (_cultureInfo != null)
             {
-                var temp = new BasicServiceProvider();
-                temp.AddService(_cultureInfo);
-                innerServices = temp;
+                innerServices.AddService(_cultureInfo);
+                hasInnerServices = true;
             }
 
-            var runtimeConfig2 = new RuntimeConfig
+            if (_additionalFunctions.Any())
+            {
+                innerServices.AddService(_additionalFunctions);

```

Comment:
>innerServices.AddService(_additionalFunctions); [](http://example.com/codeflow?start=16&length=47)

This dictionary should probably be directly on RuntimeConfig, rather than in the service provider.

Reply:
This is all internal, so Ok to handle in a future PR.

Reply:
Let's defer to a future PR